z3/src/z3_solver.cpp

```diff
@@ -319,13 +382,26 @@ Result Z3Solver::check_sat_assuming_set(const UnorderedTermSet & assumptions)
   return check_sat_assuming(z3assumps);
 }
 
-void Z3Solver::push(uint64_t num) { slv.push(); }
+void Z3Solver::push(uint64_t num)
+{
+  for (int i = 0; i < num; i++)
+  {
+    slv.push();
+  }
+}
 
 void Z3Solver::pop(uint64_t num) { slv.pop(num); }
 
 Term Z3Solver::get_value(const Term & t) const
 {
-  throw NotImplementedException("Get value not implemented for Z3 backend.");
+  shared_ptr<Z3Term> zterm = static_pointer_cast<Z3Term>(t);
+  if (zterm->is_function)
+  {
+    throw IncorrectUsageException("Cannot evaluate a function.");
+  }
+  z3::model model = slv.get_model();
+  expr eval = model.eval(zterm->term, true);
```

Comment:
Just for my own understanding, what is the second parameter here? I saw that it's `model_completion`, but what does that mean?